File: driver-service/src/main/java/vn/logistic/driver/model/Driver.java
package vn.logistic.driver.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import vn.logistic.driver.common.DriverStatus;

import java.math.BigDecimal;
import java.util.Date;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "tbl_driver")
public class Driver {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "employee_code", unique = true, nullable = false)
    private String employeeCode;

    @Column(name = "full_name", nullable = false)
    private String fullName;

    @Column(unique = true)
    private String phone;

    private String email;

    @Column(name = "license_no", unique = true)
    private String licenseNo;

    @Column(name = "license_type")
    private String licenseType;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private DriverStatus status;

    private String zone;

    private BigDecimal rating;

    @Column(name = "total_deliveries")
    private Integer totalDeliveries;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    @Temporal(TemporalType.TIMESTAMP)
    private Date createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    @Temporal(TemporalType.TIMESTAMP)
    private Date updatedAt;
}

File: frontend/src/app/core/models/driver.ts
export interface Driver {
  id?: number;
  employeeCode: string;
  fullName: string;
  phone: string;
  email?: string;
  licenseNo?: string;
  licenseType?: string;
  status: DriverStatus;
  zone?: string;
  rating?: number;
  totalDeliveries?: number;
  createdAt?: Date;
}

export enum DriverStatus {
  AVAILABLE = 'AVAILABLE',
  BUSY = 'BUSY',
  OFFLINE = 'OFFLINE',
  SUSPENDED = 'SUSPENDED'
}
